src/providers/google-vertex-ai/api.ts

```diff
@@ -0,0 +1,42 @@
+import { ProviderAPIConfig } from '../types';
+
+// Good reference for using REST: https://cloud.google.com/vertex-ai/generative-ai/docs/start/quickstarts/quickstart-multimodal#gemini-beginner-samples-drest
+// Difference versus Studio AI: https://cloud.google.com/vertex-ai/docs/start/ai-platform-users
+export const GoogleApiConfig: ProviderAPIConfig = {
+  getBaseURL: ({ providerOptions }) => {
+    const { vertexProjectId, vertexRegion } = providerOptions;
+
+    return `https://${vertexRegion}-aiplatform.googleapis.com/v1/projects/${vertexProjectId}/locations/${vertexRegion}/publishers/google`;
+  },
+  headers: ({ providerOptions }) => {
+    const { apiKey } = providerOptions;
+
+    return {
+      'Content-Type': 'application/json',
+      Authorization: `Bearer ${apiKey}`,
+    };
+  },
+  getEndpoint: ({ fn, gatewayRequestBody }) => {
+    let mappedFn = fn;
+    const { model, stream } = gatewayRequestBody;
+    if (stream) {
+      mappedFn = `stream-${fn}`;
+    }
+    switch (mappedFn) {
+      case 'chatComplete': {
+        return `/models/${model}:generateContent`;
+      }
+      case 'stream-chatComplete': {
+        return `/models/${model}:streamGenerateContent`;
```

Comment:
Vertex allows to send `?alt=sse` in url to get SSE instead of json stream. I think that will make things easy for us. Should we make that change?